simd/src/Kokkos_Simd_AVX512.hpp

```diff
@@ -0,0 +1,928 @@
+    return reinterpret_cast<value_type*>(&m_value)[i];
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION value_type operator[](std::size_t i) const {
+    return reinterpret_cast<value_type const*>(&m_value)[i];
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator*(simd const& other) const {
+    return simd(_mm256_mullo_epi32(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator+(simd const& other) const {
+    return simd(_mm256_add_epi32(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator-(simd const& other) const {
+    return simd(_mm256_sub_epi32(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator-() const {
+    return simd(0) - *this;
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION void copy_to(value_type* ptr,
+                                                element_aligned_tag) const {
+    _mm256_mask_storeu_epi32(ptr, static_cast<__mmask8>(mask_type(true)),
+                             m_value);
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION void copy_from(value_type const* ptr,
+                                                  element_aligned_tag) {
+    m_value = _mm256_mask_loadu_epi32(
+        _mm256_set1_epi32(0), static_cast<__mmask8>(mask_type(true)), ptr);
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION constexpr explicit operator __m256i() const {
+    return m_value;
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator<(simd const& other) const {
+    return mask_type(_mm256_cmplt_epi32_mask(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator>(simd const& other) const {
+    return mask_type(_mm256_cmplt_epi32_mask(other.m_value, m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator<=(simd const& other) const {
+    return mask_type(_mm256_cmple_epi32_mask(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator>=(simd const& other) const {
+    return mask_type(_mm256_cmple_epi32_mask(other.m_value, m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator==(simd const& other) const {
+    return mask_type(_mm256_cmpeq_epi32_mask(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator!=(simd const& other) const {
+    return mask_type(_mm256_cmpneq_epi32_mask(m_value, other.m_value));
+  }
+};
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<std::int32_t, simd_abi::avx512_fixed_size<8>> condition(
+    simd_mask<std::int32_t, simd_abi::avx512_fixed_size<8>> const& a,
+    simd<std::int32_t, simd_abi::avx512_fixed_size<8>> const& b,
+    simd<std::int32_t, simd_abi::avx512_fixed_size<8>> const& c) {
+  return simd<std::int32_t, simd_abi::avx512_fixed_size<8>>(
+      _mm256_mask_blend_epi32(static_cast<__mmask8>(a), static_cast<__m256i>(c),
+                              static_cast<__m256i>(b)));
+}
+
+template <>
+class simd<std::uint32_t, simd_abi::avx512_fixed_size<8>> {
+  __m256i m_value;
```

Comment:
Because it is `avx512_fixed_size<8>`, so there must be 8 of the `value_type`, so it is a 256-bit register to store 8 32-bit integers. This is the common case of folks using 8 32-bit integers to do scatter or gather operations on 8 64-bit doubles.